torch/ao/quantization/backend_config/onednn.py

```diff
@@ -0,0 +1,152 @@
+            fused_linear = nn.utils.fusion.fuse_linear_bn_eval(linear, bn)
+            fm = fused_module(fused_linear, leaky_relu)
+            return fm
+        else:
+            raise NotImplementedError("Cannot fuse eval modules: {}".format((linear, bn, leaky_relu)))
+
+# ======================
+# |  CONFIGS FOR CONV  |
+# ======================
+
+conv_configs = _get_conv_configs(conv_dtype_configs)
+
+
+# ========================
+# |  CONFIGS FOR LINEAR  |
+# ========================
+
+observation_type = ObservationType.OUTPUT_USE_DIFFERENT_OBSERVER_AS_INPUT
+linear_configs = _get_linear_configs(linear_dtype_configs)
+
+# (1) Linear + leaky_relu
+# -------------------
+# 1.1 linear module + leaky_relu fusion config
+# linear leaky_relu, linear module + leaky_relu module
+linear_configs.append(
+    BackendPatternConfig((nn.LeakyReLU, nn.Linear))
+        .set_dtype_configs(linear_dtype_configs)  # noqa: E131
+        .set_fuser_method(_reverse_sequential_wrapper2(nni.LinearLeakyReLU))
+        .set_fused_module(nni.LinearLeakyReLU))
+# linear leaky_relu, linear module + functional leaky_relu
+linear_configs.append(
+    BackendPatternConfig((F.leaky_relu, nn.Linear))
+        .set_dtype_configs(linear_dtype_configs)  # noqa: E131
+        .set_fuser_method(_reverse_sequential_wrapper2(nni.LinearLeakyReLU))
+        .set_fused_module(nni.LinearLeakyReLU))
+# linear leaky_relu, linear module + BN + leaky_relu
+linear_configs.append(
+    BackendPatternConfig((nn.LeakyReLU, (nn.BatchNorm1d, nn.Linear)))
+        .set_dtype_configs(linear_dtype_configs)  # noqa: E131
+        .set_fuser_method(_reverse3(_fuse_linear_bn_leaky_relu))
+        .set_fused_module(nni.LinearLeakyReLU))
+
+# 1.2 linear module + leaky_relu, fused module configs
+# linear leaky_relu, fused module
+linear_configs.append(
+    BackendPatternConfig(nni.LinearLeakyReLU)
+        .set_observation_type(observation_type)  # noqa: E131
+        .set_dtype_configs(linear_dtype_configs)
+        .set_root_module(nn.Linear)
+        .set_reference_quantized_module(nnqr.Linear))
+# 1.3 functional linear + leaky_relu configs
+# linear leaky_relu, functional linear + leaky_relu module
+linear_configs.append(
+    BackendPatternConfig((nn.LeakyReLU, F.linear))
+        .set_observation_type(observation_type)  # noqa: E131
+        .set_dtype_configs(linear_dtype_configs))
+# linear leaky_relu, functional linear + functional leaky_relu
+linear_configs.append(
+    BackendPatternConfig((F.leaky_relu, F.linear))
+        .set_observation_type(observation_type)  # noqa: E131
+        .set_dtype_configs(linear_dtype_configs))
+
+# =====================
+# |  BACKEND CONFIGS  |
+# =====================
+
+def get_onednn_backend_config() -> BackendConfig:
+    """
+    Return the `BackendConfig` for PyTorch's native ONEDNN backend.
+    """
+    return BackendConfig("onednn") \
+        .set_backend_pattern_configs(conv_configs) \
+        .set_backend_pattern_configs(linear_configs)
```

Comment:
For other ops, do I need to copy default configs here or just ignore them? https://github.com/pytorch/pytorch/pull/88665#discussion_r1020668935

Reply:
Looks like they are needed. I have added them back. Please take a look again. Thanks.

Reply:
OK looks good, so for the other ops do we just fallback to the default (fbgemm) implementation?

Reply:
Yes, we use default native implementations for ops other than conv/linear. However, we still need to set pattern configs for other ops here. Otherwise, those ops are not quantized if we use onednn's backend config for `prepare_fx` and `convert_fx`.